pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -495,16 +495,26 @@ func (lb *ListenerBuilder) buildSidecarOutboundListeners(node *model.Proxy,
 							// Make sure each endpoint address is a valid address
 							// as service entries could have NONE resolution with label selectors for workload
 							// entries (which could technically have hostnames).
-							if !netutil.IsValidIPAddress(instance.Address) {
+							isValidInstance := true
+							for _, addr := range instance.Addresses {
+								if !netutil.IsValidIPAddress(addr) {
+									isValidInstance = false
+									break
+								}
+							}
+							if !isValidInstance {
 								continue
 							}
 							// Skip build outbound listener to the node itself,
 							// as when app access itself by pod ip will not flow through this listener.
 							// Simultaneously, it will be duplicate with inbound listener.
-							if instance.Address == node.IPAddresses[0] {
+							nodeAddrs := node.IPAddresses
+							sort.Sort(sort.Reverse(sort.StringSlice(nodeAddrs)))
```

Comment:
This is racy and not an appropriate place to sort these. Its already an ordered list _and the order matters and is well defined_.

Reply:
Hi @howardjohn, yeah, this has been change based on above comment. Unnecessary to sort them.